site/src/routes/metrics.js

```diff
@@ -0,0 +1,154 @@
+import { stores } from '../constants.js'
+import { get as getDeals } from '../models/deals.js'
+
+const MAX_AGE_SECS = 60 * 10 // max age of a metrics response in seconds
+
+/**
+ * TODO: basic auth
+ * @param {FetchEvent} event
+ */
+export async function metrics(event) {
+  const cacheKey = new Request(new URL(event.request.url).toString(), event.request)
+  const cache = caches.default
+
+  let res = await cache.match(cacheKey)
+  if (res) return res
+
+  const [userMetrics, nftMetrics] = await Promise.all([getUserMetrics(), getNftMetrics()])
+  res = new Response(exportPromMetrics({ userMetrics, nftMetrics }))
+  // Cache the response for 60s
+  res.headers.append('Cache-Control', `s-maxage=${MAX_AGE_SECS}`)
```

Comment:
```suggestion
  res.headers.append('Cache-Control', `maxage=${MAX_AGE_SECS}`)
```

no need to you restrict the cache to just cache proxies, like this browser and CF will cache it

Reply:
Actually here i would do `public,max-age=600,stale-while-revalidate=3600`

Reply:
this will give us more certainty that we get fresh values in the metrics at least every hour